features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/TouchPrivacyManager.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal
+
+import android.graphics.Point
+import android.graphics.Rect
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.sessionreplay.TouchPrivacy
+
+internal class TouchPrivacyManager(
+    private val globalTouchPrivacy: TouchPrivacy
+) {
+    // areas on screen where overrides are applied
+    private val currentOverrideAreas = HashMap<Rect, TouchPrivacy>()
+
+    // built during the view traversal and copied to currentOverrideSnapshot at the end
+    private val nextOverrideAreas = HashMap<Rect, TouchPrivacy>()
+
+    @UiThread
+    internal fun addTouchOverrideArea(bounds: Rect, touchPrivacy: TouchPrivacy) {
+        nextOverrideAreas[bounds] = touchPrivacy
+    }
+
+    @UiThread
+    internal fun updateCurrentTouchOverrideSnapshot() {
+        currentOverrideAreas.clear()
+        // NPE cannot happen here
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        currentOverrideAreas.putAll(nextOverrideAreas)
+        nextOverrideAreas.clear()
+    }
+
+    @UiThread
+    internal fun shouldRecordTouch(touchLocation: Point): Boolean {
+        var isOverriddenToShowTouch = false
+
+        // Everything is UiThread, so ConcurrentModification cannot happen here
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        currentOverrideAreas.forEach { entry ->
+            val area = entry.key
+            val overrideValue = entry.value
+
+            if (isWithinOverrideArea(touchLocation, area)) {
+                when (overrideValue) {
+                    TouchPrivacy.HIDE -> return false
+                    TouchPrivacy.SHOW -> isOverriddenToShowTouch = true
+                }
+            }
+        }
+
+        return if (isOverriddenToShowTouch) true else globalTouchPrivacy == TouchPrivacy.SHOW
+    }
+
+    @VisibleForTesting
+    internal fun getCurrentOverrideSnapshot(): Map<Rect, TouchPrivacy> {
+        return currentOverrideAreas
+    }
+
+    @VisibleForTesting
+    internal fun getNextOverrideSnapshot(): Map<Rect, TouchPrivacy> {
+        return nextOverrideAreas
+    }
+
+    private fun isWithinOverrideArea(touchPoint: Point, overrideArea: Rect): Boolean =
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ I think here you can user `overrideArea.contains(x,y)` right ?